interactive/dashboard/flow_diagram.py

```diff
@@ -119,153 +94,1008 @@ def name_included_reactions(included_reactions, reactions_json):
+        for species in list_of_species:
+            species_colors.update({species: "#6b6bdb"})
+            species_sizes.update({species: 40})
+            if 'reactants' in r:
+                if species in r['reactants']:
+                    tmp_val = reactions_data['camp-data'][0]['reactions']
+                    contained_reactions.update(
+                        {tmp_val.index(r): {}})
+                    reaction_string = ""
+                    for reactant in r['reactants']:
+                        species_nodes.update({reactant: {}})
+                        reaction_string = reaction_string + reactant + "_"
+                    # remove last _ and replace with ->
+                    reaction_string = reaction_string[:-1] + "->"
+                    for product in r['products']:
+                        species_nodes.update({product: {}})
+                        reaction_string = reaction_string + product + "_"
+                    if r['products'] == [] or len(r['products']) == 0:
+                        # reaction_string = reaction_string[:-2]
+                        print("* no products:", reaction_string)
+                    else:
+                        reaction_string = reaction_string[:-1]
+                    reactions_nodes.update({reaction_string: {}})
+                if species in r['products']:
+                    tmp_val = reactions_data['camp-data'][0]['reactions']
+                    contained_reactions.update(
+                        {tmp_val.index(r): {}})
+                    reaction_string = ""
+                    for reactant in r['reactants']:
+                        species_nodes.update({reactant: {}})
+                        reaction_string = reaction_string + reactant + "_"
+                    # remove last + and replace with →
+                    reaction_string = reaction_string[:-1] + "->"
+                    for product in r['products']:
+                        species_nodes.update({product: {}})
+                        reaction_string = reaction_string + product + "_"
+                    reaction_string = reaction_string[:-1]
+                    reactions_nodes.update({reaction_string: {}})
+    for speci in species_nodes:
+        if speci not in list_of_species:
+            species_colors.update({speci: "#94b8f8"})
+            species_sizes.update({speci: 20})
+    return (contained_reactions, species_nodes,
+            reactions_nodes, species_colors, species_sizes)
 
+
+# return list of raw widths for arrows. Also set new minAndMax
+def findReactionRates(reactions_nodes, df, start, end, csv_results_path_default=csv_results_path_default):
+    global minAndMaxOfSelectedTimeFrame
+    global userSelectedMinMax
+    global previous_vals
+    rates_cols = [x for x in df.columns if 'myrate' in x]
+    reactionsToAdd = []
+    for re in reactions_nodes:
+        # convert index of reaction into actual reaction name
+        reactionsToAdd.append(re)
```

Comment:
The comment indicates something should be converted. It appears that this is not happening and that you are making a list of the exact same size and contents of `reactions_nodes`.

Reply:
@jwoody02 update on this?